website/content/en/docs/building-operators/golang/project_migration_guide.md

```diff
@@ -1,182 +1,315 @@
+### Setting the RBAC permissions
 
-## Generate CRDs
+The RBAC permissions now are configured via [RBAC markers][rbac_markers] whcih will generate and update the manifest files present in `config/rbac/`. See that on top of the `Reconcile()` method we will find them, in our Memcached example 
+it will be: 
 
-Run [`make manifests`][generate_crd] to generate CRD manifests. They would be generated inside the `config/crd/bases` folder.
+```go
+// +kubebuilder:rbac:groups=cache.example.com,resources=memcacheds,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=cache.example.com,resources=memcacheds/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=apps,resources=deployments,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=apps,resources=services,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=core,resources=pods,verbs=get;list
+```
 
-## Configuring your test environment
+To update `config/rbac/role.yaml` after changing the markers, run `make manifests`.  Also, you might want to configure additional permissions for your project, see [RBAC markers][rbac_markers] doc. 
 
-Projects are scaffolded with unit tests that utilize the [envtest](https://godoc.org/sigs.k8s.io/controller-runtime/pkg/envtest)
-library, which requires certain Kubernetes server binaries be present locally.
-Installation instructions can be found [here][env-test-setup].
+By default, the projects are `cluster-scoped`. Check that your role is of the `ClusterRole` Kind. For further information check the [operator-scope][operator-scope] doc. 
 
-## Operator Manifests
+**Note** You  would like to check the final result of our [memcached_controller.go][memcached_controller] migrated to the new layout. 
 
-### Operator deployment manifests
+## Migrating the Main.go
 
-The operator deployment manifest [`deploy/operator.yaml`][deployment_yaml] from the old project should be copied over to `config/manager/manager.yaml` in the new project.
+If you are using third-party API's in your project for example then, it means that you need to copy your customizations 
+from `cmd/manager/main.go` to `main.go`. You’ll also need to ensure all of the needed schemes have been registered.
 
-**Note**:
-The kustomize file requires the operator deployment manifest to have the field `namespace` which is missing in the `deploy/operator.yaml` manifest of SDK project.
-> **// TODO:** Explain label propagation for metrics collection.
+By checking our new `main.go` we will find that:
 
-### RBAC permissions
+- The SDK `leader.Become` was replaced by the [leaderElection][leaderElection]. However, we still able to use the 
+previous one as well by adding the [operator-lib][operator-lib] as dependency fo the project.
+- The default port used by the metric endpoint binds to from `:8383` to `:8080`. To continue using port `8383`, specify `--metrics-addr=:8383` when you start the operator. 
 
```

Comment:
We also removed all the code for `addMetrics`, so we need to tell people how to get that functionality back (i.e. setup custom watches using `InstrumentedEnqueueRequestForObject` from operator-lib)

Reply:
Yep. Shows reasonable. However, I think the `InstrumentedEnqueueRequestForObject ` needs to be with the metrics. I will add this.